### PR TITLE
Fixes integration of dynamodb compatible endpoints (#3329)

### PR DESCRIPTION
**Pull request by balpert89**
This PR integrates the possibility to pass in endpoints for dynamodb-compatible endpoints.

Usage (e.g. in a Kubernetes environment, `mcs-dynamodb` is a service resource):
```
teleport:
  storage:
    audit_events_uri:
    - dynamodb://teleport?endpoint=http://mcs-dynamodb:8000
```

### CODE REVIEW

**Comment by webvictim** [comment]
ok to test

**Comment by webvictim** [comment]
```
# github.com/gravitational/teleport/lib/service
lib/service/service.go:845:5: self-assignment of endpoint to endpoint
Makefile:198: recipe for target 'test' failed
```

**Comment by webvictim** [comment]
Tests pass OK now, will leave this for the reviewers.

File: lib/events/dynamoevents/dynamoevents.go [hunk 1]
```diff
@@ -172,6 +174,11 @@ func New(cfg Config) (*Log, error) {
 		sess.Config.Region = aws.String(cfg.Region)
 	}
 
+	// override the endpoint from the URI query parameter in the YAML file
```
Comment (russjones) [inline lib/events/dynamoevents/dynamoevents.go]:
```go
// Override the service endpoint using the "endpoint" query parameter from 
// "audit_events_uri". This is for non-AWS DynamoDB-compatible backends.
```

File: lib/service/service.go [hunk 2]
```diff
@@ -840,8 +840,11 @@ func initExternalLog(auditConfig services.AuditConfig) (events.IAuditLog, error)
 			loggers = append(loggers, logger)
 		case dynamo.GetName():
 			hasNonFileLog = true
+			endpoint := uri.Query().Get(teleport.Endpoint)
```
Comment (russjones) [inline lib/service/service.go]: Can you add a `SetFromURL` method to `dynamoevents.Config` similar to `firestoreevents.EventsConfig`? Then `SetFromURL` can be used to override the endpoint value?

File: lib/events/dynamoevents/dynamoevents.go (hunk 1 above)
Comment (balpert89) [reply lib/events/dynamoevents/dynamoevents.go]: Done :)

File: lib/service/service.go (hunk 2 above)
Comment (balpert89) [reply lib/service/service.go]: Done :)

**Review by russjones — APPROVED** [review]

**Review by webvictim — APPROVED** [review]
LGTM.

**Review by fspmarshall — APPROVED** [review]

**Comment by balpert89** [comment]
@klizhentas when will this be included in a release? Found no indication in tags 4.2.8 and none in 4.2.9-beta.1 so far.

**Pull request merged**